### ISSUE TITLE
Since #132, one timed-out request makes every later one wait 30s too

### ISSUE THREAD

**Issue by jvdburgh** [sergioazoc/oxlint-tailwindcss#145]
After upgrading to 1.10.2 our CI lint went from about 90 seconds to over ten minutes, at which point the runner killed it for producing no output. When we did get output, every error was `canonicalize worker request timed out after 30000ms`, on unremarkable classes like `max-w-[500px]` and `p-[0.2rem]`.

The timeouts themselves aren't new — 1.10.0 reported the same ones on the same code. What changed is what they cost. On 1.10.0 the first one was remembered and the rest of the run failed fast. Since #132 each timeout drops the worker and retries, so every remaining class list waits the full 30 seconds again.

Here's a repro that doesn't need anything to actually be slow — a handler that never replies stands in for one that is:

```ts
import { resolve } from 'node:path'
import { DesignSystemWorker, makeWorkerScript } from '../../src/design-system/ds-worker'

const worker = new DesignSystemWorker<unknown, unknown>({
  workerScript: makeWorkerScript('(ds, req) => { while (true) {} }'),
  serviceName: 'canonicalize',
})

for (let i = 0; i < 5; i++) {
  const t = Date.now()
  try { worker.callSync(resolve(__dirname, '../fixtures/default.css'), ['flex']) } catch {}
  console.log(Date.now() - t)
}
worker.reset()
```

Set `REQUEST_TIMEOUT = 200` first so it finishes quickly. On main (34c85bd) I get `297, 320, 284, 283, 315` — five calls, five full timeouts, plus a worker respawn each time. At the shipped 30s that's two and a half minutes for five class lists. We had fifteen.

The cause is the timeout branch in `callSync`, which drops the worker and throws without `remember()`. That's clearly right for the `null` sentinel #130 was about: a malformed input really is a property of that one input, and the next one deserves a fresh try. A timeout isn't like that — a machine slow enough to time out once will time out again, so the retry is guaranteed to cost another 30 seconds.

What I can't tell you is *why* the requests are slow to begin with. It doesn't reproduce locally for us (~1.4s warm, ~29s cold, no timeouts), and it isn't just the number of arbitrary values — another project of ours has more of them and lints fine. So treat this as a report about the cost once a timeout happens, not about the cause.

I did write a patch that bounds the retry — go sticky after a few consecutive failures for the same entry point, reset on any success — but withdrew it in #144, since the sensible bound probably depends on the underlying cause and that's your call rather than mine. Happy to pick it back up in whatever shape you'd prefer.

**Comment by sergioazoc** [comment]
Thanks for the excellent report @jvdburgh, the repro and the O(files) framing nailed it.

Fixed in 1.11.0: per-request failures (timeouts included) go sticky after 3 consecutive fails per entry point, so the run fails fast instead of re-paying 30s on every remaining class list. A success clears the budget, and the sticky expires after a backoff window so a long-lived editor self-heals (no more dead-until-restart). The #130 behavior is unchanged.

I also added `OXLINT_TAILWINDCSS_WORKER_REQUEST_TIMEOUT` for your ~29s-cold case: raise it and the request completes instead of failing fast. Worth trying on those two repos to see if they even hit the bound anymore.

Thanks again for the root-cause dig and the #144 patch.
